src/main/java/seedu/module/model/ModuleManager.java

```diff
@@ -117,8 +182,41 @@ public static void clearMapping() {
      *
      * @return List of String associated with basic modules
      */
-    static List<String> initListOfModulesAccepted() {
+    private static List<String> initListOfModulesAccepted() {
         List<String> listOfModules = new ArrayList<>(Arrays.asList(arrOfModules));
         return listOfModules;
     }
+
+    /**
+     * Fills the observable list of existringModules with the keySet of the mapping.
+     */
+    public static void setExistingModuleList() {
+        List<Module> existingModules = List.copyOf(mappingOfModulesToTasks.keySet());
+        moduleList.setAll(existingModules);
+    }
+
+    /**
+     * Fills the observable list of existringModules with the keySet of the mapping.
+     */
+    public static void setModulePieChartData() {
+        modulePieChartData.clear();
+        for (int i = 0; i < moduleList.size(); i++) {
+            Module module = moduleList.get(i);
+            modulePieChartData.add(new PieChart.Data(module.toString(), moduleWorkLoadDistribution.get(module)));
```

Comment:
Minor suggestion: can use for-each loop instead but no biggie

Reply:
okay! Do you think this is better?
```Java
modulePieChartData.setAll(moduleList.stream().map(module -> new PieChart.Data(module.toString(),
            moduleWorkLoadDistribution.get(module))).collect(Collectors.toList()))
```

Reply:
If it works, I guess it can be used